lib/segment/src/index/field_index/btree_index.rs

```diff
@@ -105,6 +105,36 @@ impl<T: KeyEncoder> NumericIndex<T> {
         })?;
         Ok(db_ref.flush_cf(cf_handle)?)
     }
+
+    pub fn remove_point(&self, id: PointOffsetType) -> OperationResult<()> {
+        let db_ref = self.db.borrow();
+        let cf_handle = db_ref.cf_handle(&self.store_cf_name).ok_or_else(|| {
+            OperationError::service_error(&format!(
+                "Index flush error: column family {} not found",
+                self.store_cf_name
+            ))
+        })?;
+
+        let mut iter = db_ref.raw_iterator_cf(&cf_handle);
+
+        iter.seek_to_first();
+
+        while iter.valid() {
```

Comment:
Do we iterate over the whole DB to delete one point? Maybe we can use range search with key prefix?

Reply:
yes, we should use the full key ( value + offset ) to avoid the full scan, but in order to use the full key we need to refactor the api. In the same commit I left a comment about this behaviour here https://github.com/qdrant/qdrant/pull/642/files#diff-eb69fa5c53aae002e7a33e63f74edea0cbb87c9d9ab35201a0f6f61cf066811cR62

Reply:
In merged PR about histogram, I added id to values map. We can use this map for getting all key+value pairs